File: src/main/scala/com/xmlcalabash/util/VoidLocation.scala
package com.xmlcalabash.util

import net.sf.saxon.s9api.Location

// Via Florent Georges

object VoidLocation {
  private val loc = new VoidLocation()
  def instance(): VoidLocation = loc
}

class VoidLocation extends Location {
  override def getSystemId: String = null

  override def getPublicId: String = null

  override def getLineNumber: Int = -1

  override def getColumnNumber: Int = -1

  override def saveLocation(): Location = this
}

File: xmlcalabash/src/main/kotlin/com/xmlcalabash/util/VoidLocation.kt
package com.xmlcalabash.util

import net.sf.saxon.s9api.Location

class VoidLocation private constructor(): Location {
    companion object {
        val instance = VoidLocation()
    }

    override fun getPublicId(): String? = null
    override fun getSystemId(): String? = null
    override fun getLineNumber(): Int = -1
    override fun getColumnNumber(): Int = -1
    override fun saveLocation(): Location = this
}